ui: eslint should error on eqeqeq violation

xinhaoz opened the pull request:
Add eqeqeq eslint rule for cluster-ui. This rule is on 'smart' mode, meaning we must use triple equals unless:
- comparing two literal values
- evaluating the value of typeof
- comparing against null

Epic: none

Release note: None

xinhaoz committed:
cluster-ui: eslint should error on eqeqeq violation

Add eqeqeq eslint rule for cluster-ui. This rule is on
'smart' mode, meaning we must use triple equals unless:
- comparing two literal values
- evaluating the value of typeof
- comparing against null

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111445)
<!-- Reviewable:end -->

xinhaoz committed:
db-console: eslint should error on eqeqeq violation

Add eqeqeq eslint rule for db-console. This rule is on
'smart' mode, meaning we must use triple equals unless:
- comparing two literal values
- evaluating the value of typeof
- comparing against nul

Epic: none

Release note: None

maryliag reviewed (approved):
Nice!
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 32 of 32 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111445#-:-NfS_EE464toOUWQBrKR:bkjg46h)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)



<!-- Sent from Reviewable.io -->

xinhaoz commented:
bors r+
